modules/form-field/react/lib/FormField.tsx

```diff
@@ -106,7 +104,7 @@ export default class FormField extends React.Component<FormFieldProps> {
         }
       }
 
-      if (!child.props.id && this.props.inputId) {
+      if (this.props.inputId) {
```

Comment:
The way this was setup before you could have a label and its input have mismatch ids
```
<FormElement inputId='foo'><TextInput id='bar' /></FormElement>
```
=>
```
<label for='foo' />
<input id='bar />
```